Week2/hw-part1.js

```diff
@@ -0,0 +1,137 @@
+const prompts = require('prompts');
+const mysql = require('mysql');
+const util = require('util');
+const colors = require('colors');
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'new_world',
+  multipleStatements: false,
+});
+
+const execQuery = util.promisify(connection.query.bind(connection));
+
+const questions = [
+  {
+    type: 'text',
+    name: 'CountryCode',
+    message: 'Q1. Enter a country code to see its capital',
+    initial: 'Example: NLD',
+  },
+  {
+    type: 'text',
+    name: 'region',
+    message: 'Q2. Enter a region to find the languages',
+    initial: 'Example: South America',
+  },
+  {
+    type: 'text',
+    name: 'language',
+    message: 'Q3. Enter a language to find the number of cities it is spoken',
+    initial: 'Example: English',
+  },
+  {
+    type: 'text',
+    name: 'region2',
+    message: 'Q4. Enter a region ',
+    initial: 'Example: Middle East',
+  },
+  {
+    type: 'text',
+    name: 'language2',
+    message: 'Q4. Enter a language',
+    initial: 'Example: Arabic',
+  },
+  {
+    type: 'text',
+    name: 'continent',
+    message: 'To get continents with number of languages spoken',
+    initial: 'Just press enter',
+  },
+];
+
+async function queryDatabase() {
+  let inputCountryCode = '';
+  let inputRegion = '';
+  let inputLanguage = '';
+  let inputRegOf = '';
+  let inputLangOf = '';
+
+  try {
+    const onSubmit = (prompt, response) => console.log(`You typed ${colors.rainbow(response)}`);
+    const response = await prompts(questions, { onSubmit });
+
+    inputCountryCode = response.CountryCode.toUpperCase();
+    inputRegion = response.region
+      .toLowerCase()
+      .split(' ')
+      .map(s => s.charAt(0).toUpperCase() + s.substring(1))
+      .join(' ');
+    inputLanguage = response.language.replace(/^\w/, c => c.toUpperCase());
+    inputRegOf = response.region2
+      .toLowerCase()
+      .split(' ')
+      .map(s => s.charAt(0).toUpperCase() + s.substring(1))
+      .join(' ');
+    inputLangOf = response.language2.replace(/^\w/, c => c.toUpperCase());
+
+    const selectCapital =      'SELECT name FROM city WHERE ID =(SELECT capital FROM country WHERE Code = ?);';
+    const selectLang =      'SELECT Language FROM countrylanguage JOIN country ON CountryCode = Code WHERE region = ?';
+    const selectNum = 'SELECT count(countryCode) FROM countrylanguage WHERE Language = ?';
+    const selectOfficial = `SELECT countryCode, name AS Country, isOfficial FROM country JOIN countryLanguage ON countryCode = Code WHERE region = '${inputRegOf}' AND language = '${inputLangOf}'`;
```

Comment:
I think you misunderstood the question in the homework. You print False if you don't find any official language. You have printed False if the language is not official. So just add the isOfficial condition in your where statement and it's good :)

Reply:
Thanks! I actually understood the question right but I couldn't make it work, so I did the next best thing:) But now it works and I am happy thank you